erpnext/accounts/report/profitability_analysis/profitability_analysis.py

```diff
@@ -60,7 +62,9 @@ def get_data(accounts, filters, based_on):
 		filters.get("company"),
 		filters.get("from_date"),
 		filters.get("to_date"),
-		based_on,
+		based_on
+		if based_on != "accounting_dimension"
+		else filters.accounting_dimension.replace(" ", "_").lower(),
```

Comment:
Use fieldname from the Accounting Dimension doctype instead of converting to lowercase